File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/DailyForecastResponse.java
package io.github.tobyhs.weatherweight.api.accuweather.forecasts;

import java.util.List;

import com.google.auto.value.AutoValue;
import com.google.gson.annotations.SerializedName;
import me.tatarka.gsonvalue.annotations.GsonBuilder;

/**
 * A response from one of AccuWeather's Forecasts API endpoints
 */
@AutoValue
public abstract class DailyForecastResponse {
    /**
     * @return list of daily forecasts
     */
    @SerializedName("DailyForecasts")
    public abstract List<DailyForecast> getDailyForecasts();

    @GsonBuilder
    public static Builder builder() {
        return new AutoValue_DailyForecastResponse.Builder();
    }

    @AutoValue.Builder
    public static abstract class Builder {
        public abstract Builder setDailyForecasts(List<DailyForecast> dailyForecasts);
        public abstract DailyForecastResponse build();
    }
}

File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/DailyForecastResponse.kt
package io.github.tobyhs.weatherweight.api.accuweather.forecasts

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * A response from one of AccuWeather's Forecasts API endpoints
 *
 * @property dailyForecasts list of daily forecasts
 */
@JsonClass(generateAdapter = true)
data class DailyForecastResponse(
    @Json(name = "DailyForecasts") val dailyForecasts: List<DailyForecast>
)
